### DISCUSSION TITLE
Use js hooks to prevent record being updated (0.23)

### DISCUSSION THREAD

**Discussion by andrewpmoore** [pocketbase/pocketbase#6101] [Q&A]
I have a collection called `customer_store`, it has a Date field called `updated_at`.

I'm wanting that if the incoming update to that record has an `updated_at` time before the record in the database that the record is rejected.

I've got some other hooks working ok, but I feel like this should be simple, but I'm not sure how to do it.
Is there a way of getting the `new` data and the `before` data to do a comparison and then reject based on that?

**Comment by ganigeorgiev** [comment]
How is this `updated_at` expected to be populated - programmatically by the hook or via client submitted value?

If you want to perform the check against the client submitted value then you can use the collection Update API rule for this with requirement like: `updated_at < @request.body.updated_at`.

If you want to do it with a hook, then you can attach either to the [record model hooks](https://pocketbase.io/docs/js-event-hooks/#record-model-hooks) or to the [record request hooks](https://pocketbase.io/docs/js-event-hooks/#request-hooks) (the latter gives you access to the request context, aka. the authenticated user, headers, etc.). For example:

```js
onRecordUpdateRequest((e) => {
    if (e.hasSuperuserAuth())  {
        return e.next() // ignore for superusers to allow setting any date
    }

    // see https://pocketbase.io/docs/js-records/#copies
    let original = e.record.original();

    if (original.getDateTime("updated_at").after(e.record.getDateTime("updated_at"))) {
        throw new BadRequestError("updated_at must be after " + original.getDateTime("updated_at").string())
    }

    e.next()
}, "customer_store")
```

**Reply by andrewpmoore** [reply]
Thanks, yes it's from a client and that works perfectly. 
Thank you for pocketbase and the very quick reply 🙂
